sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosBulkTest.java

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.models.PartitionKey;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosBulkTest extends BatchTestBase {
+
+    private CosmosAsyncClient bulkClient;
+    private CosmosAsyncContainer bulkAsyncContainer;
+
+    @Factory(dataProvider = "clientBuildersWithDirectSession")
+    public CosmosBulkTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosBulkTest() {
+        assertThat(this.bulkClient).isNull();
+        this.bulkClient = getClientBuilder().buildAsyncClient();
+        bulkAsyncContainer = getSharedMultiPartitionCosmosContainer(this.bulkClient);
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeCloseAsync(this.bulkClient);
+    }
+
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void createItem_withBulk() {
+        int totalRequest = getTotalRequest();
+
+        Flux<CosmosItemOperation> cosmosItemOperationFlux = Flux.merge(
+            Flux.range(0, totalRequest).map(i -> {
+                String partitionKey = UUID.randomUUID().toString();
+                TestDoc testDoc = this.populateTestDoc(partitionKey);
+
+                return BulkProcessingUtil.getBulkCreateItemOperation(testDoc, new PartitionKey(partitionKey));
+            }),
+            Flux.range(0, totalRequest).map(i -> {
+                String partitionKey = UUID.randomUUID().toString();
+                EventDoc eventDoc = new EventDoc(UUID.randomUUID().toString(), 2, 4, "type1", partitionKey);
+
+                return BulkProcessingUtil.getBulkCreateItemOperation(eventDoc, new PartitionKey(partitionKey));
+            }));
+
+        BulkProcessingOptions<CosmosBulkTest> bulkProcessingOptions = new BulkProcessingOptions<>();
+        bulkProcessingOptions.setMaxMicroBatchSize(100);
+        bulkProcessingOptions.setMaxMicroBatchConcurrency(5);
+
+        Flux<CosmosBulkOperationRequestAndResponse<CosmosBulkTest>> responseFlux = bulkAsyncContainer
+            .processBulkOperations(cosmosItemOperationFlux, bulkProcessingOptions);
```

Comment:
could you please also add tests for case when the input flux `cosmosItemOperationFlux` errors.

Reply:
Have added one Mo and will make it better. Right now we are eating up the errors passed, and I cannot find another way. I wanted to have a response for it too, with the operation/response null and user passed exception in CosmosBulkOperationResponse. Is ignoring the error signal valid way?